x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/DocumentSubsetBitsetCache.java

```diff
@@ -178,7 +179,7 @@ private void onCacheEviction(RemovalNotification<BitsetCacheKey, BitSet> notific
                 // it's possible for the key to be back in the cache if it was immediately repopulated after it was evicted, so check
                 if (bitsetCache.get(bitsetKey) == null) {
                     // key is no longer in the cache, make sure it is no longer in the lookup map either.
-                    keysByIndex.getOrDefault(indexKey, Set.of()).remove(bitsetKey);
+                    keysByIndex.getOrDefault(indexKey, Collections.emptySet()).remove(bitsetKey);
```

Comment:
I don't think this change justifies a new test. But please let me know if you think otherwise. Thanks!

Reply:
My fear is that someone will decide to do a refactor of `Collections.emptySet()` back to `Set.of()` and nothing will detect the problem.
I think we need something that explains or tests that this cannot be an ImmutableSet.

Reply:
Fair enough. I added a test for it and also changed the fix to handle `null` value explicitly instead of relying on emptySet.